### DISCUSSION TITLE
Unable to Use Serial2 and Serial3

### DISCUSSION THREAD

**Discussion by Shubham916701** [stm32duino/Arduino_Core_STM32#1855] [Q&A] [answered]
Hello,

I have been using the Arduino IDE for programming controllers from Atmel, esp, and teensy. 
Now trying my hands on STM32 Blue Pill. 

I have an application for which I will need to use the 3 Hardware Serial Ports of the Bluepill.
Am using this image for reference. 
![image](https://user-images.githubusercontent.com/93254458/198834625-675ee0ab-bbf5-421d-a655-5e95dae2e8a5.png)
[https://lh3.googleusercontent.com/YxZFvsHzIxhZSfSpUbmADkKbx6mwlExshpYgs4anPNOnYkqsS9kp_JdcWuecbrtlW8vNSP9jKElkvYGaOBZWNlRFAbZnaJZCIfaqqEVt7s7BccuJrthHiCRHXVz-HpqFAKAYhT1UsVWsFwtnaQ](URL)

The Board selection is shown below. 
![image](https://user-images.githubusercontent.com/93254458/198834744-b8b66813-df88-4ca5-abb2-23231bb1166a.png)

When I use Serial.begin(9600) or Serial1.begin(9600), the code gets compiled without any errors. 
Moving ahead, when I use Serial2.begin(9600), the compilation ends with errors as shown here. 
![image](https://user-images.githubusercontent.com/93254458/198835053-8102fa01-da3a-45b1-8fb2-5f83a84ef1fb.png)

The error occurs when i try to use Serial2. 
I need help in fixing what am doing wrong here. 

Looking forward to a Saviour.

**Comment by fpistm** [comment, marked as answer]
Hi @Shubham916701 
You can refer to https://github.com/stm32duino/wiki/wiki/API#hardwareserial

As stated: 

> By default, only one Serialx instance is available mapped to the generic Serial name.

For BluePill by default `Serial` instance is mapped to `Serial1`:
https://github.com/stm32duino/Arduino_Core_STM32/blob/4e9dae5c87b5f5331545b02c0fa51d97228e4a37/variants/STM32F1xx/F103C4T_F103C6(T-U)/variant_BLUEPILL_F103C6.h#L125-L137

To use `Serial2`, you have to declare them:
`HardwareSerial Serial2(PA3, PA2);`

Anyway STM32F103C6 has only 2 USART not 3. Maybe you got a BluePill F103C8?
In that case:
`HardwareSerial Serial2(PA3, PA2);`
`HardwareSerial Serial2(PB11, PB10);`

**Reply by Shadrack-N** [reply]
awesome, thanks
